shakenfist/external_api/base.py

```diff
@@ -68,16 +72,37 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def valid_uuid4(uuid_string):
+    try:
+        uuid.UUID(uuid_string, version=4)
+    except ValueError:
+        return False
+    return True
+
+
 def arg_is_instance_uuid(func):
     # Method uses the instance from the db
     def wrapper(*args, **kwargs):
-        if 'instance_uuid' in kwargs:
-            kwargs['instance_from_db'] = Instance.from_db(
-                kwargs['instance_uuid'])
-        if not kwargs.get('instance_from_db'):
-            LOG.with_instance(kwargs['instance_uuid']).info(
-                'Instance not found, genuinely missing')
-            return error(404, 'instance not found')
+        with ThreadLocalReadOnlyCache():
+            inst_uuid = kwargs.get('instance_uuid')
+            if inst_uuid and not valid_uuid4(inst_uuid):
+                # Check if uuid is a valid name of an instance
+                filters = [
+                    partial(baseobject.namespace_filter, get_jwt_identity()[0]),
+                    instance.active_states_filter,
+                ]
+                for i in instance.Instances(filters):
```

Comment:
This feels inefficient to me -- you have an instance object, and then you store its uuid and look it up again below. That said, I not its inside a cache block.

Do you think its worth re-writing this to have `instance=None` on line 87, and then only do an instance lookup if instance is still None on like 98?

Reply:
Good point. Will sort out.